src/testdrive/src/action/kinesis/update_shards.rs

```diff
@@ -58,41 +58,50 @@ impl Action for UpdateShardCountAction {
             .map_err(|e| format!("adding shards to stream {}: {}", &stream_name, e))?;
 
         // Verify the current shard count.
-        retry::retry_for(Duration::from_secs(60), |_| async {
-            // Wait for shards to stop updating.
-            let description = state
-                .kinesis_client
-                .describe_stream(DescribeStreamInput {
-                    exclusive_start_shard_id: None,
-                    limit: None,
-                    stream_name: stream_name.clone(),
-                })
-                .await
-                .map_err(|e| format!("getting current shard count: {}", e))?
-                .stream_description;
-            if description.stream_status != "ACTIVE" {
-                return Err(format!(
-                    "stream {} is not active, is {}",
-                    stream_name, description.stream_status
-                ));
-            }
+        retry::retry_for(
+            Duration::from_secs_f64(if state.default_timeout > 60.0 {
+                state.default_timeout
+            } else {
+                60.0
+            }),
```

Comment:
This pattern can be more succinctly expressed here and throughout like so!

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a73a825af2f49dee33371d70c3294788

Reply:
Yes, I tried that initially but unfortunately ```cmp``` does not work over ```f64``` because ```f64``` does not have the ```PartialEq``` trait . Will try to pass two ```Duration``` objects to ```cmp``` instead.